pkg/operator/rules_test.go

```diff
@@ -0,0 +1,234 @@
+package operator
+
+import (
+	"testing"
+
+	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_generateRules(t *testing.T) {
```

Comment:
nit: we typically use the [recommended](https://go.dev/doc/effective_go#mixed-caps) `camelCase` over `snake_case` in our repo.